### PR TITLE
zephyr: Use Zephyr provided definitions for images/slots

### PR DESCRIPTION
**Pull request by de-nordic**
The commit moves from directly accessing images/slots with FIXED_PARTITION macros and DTS labels to definitions provided by zephyr/dfu/mcuboot_partitions.h.

Signed-off-by: Dominik Ermel <dominik.ermel@nordicsemi.no>

### CODE REVIEW

File: boot/zephyr/include/sysflash/sysflash.h [hunk 1]
```diff
@@ -45,8 +53,10 @@
 
 #else /* CONFIG_SINGLE_APPLICATION_SLOT */
```
Comment (nordicjm) [inline boot/zephyr/include/sysflash/sysflash.h:54]:
```suggestion
#else /* !CONFIG_SINGLE_APPLICATION_SLOT */
```

File: boot/zephyr/include/sysflash/sysflash.h [hunk 2]
```diff
@@ -45,8 +53,10 @@
...
 
-#define FLASH_AREA_IMAGE_PRIMARY(x)	FIXED_PARTITION_ID(slot0_partition)
-#define FLASH_AREA_IMAGE_SECONDARY(x)	FIXED_PARTITION_ID(slot0_partition)
+#define FLASH_AREA_IMAGE_PRIMARY(x)                             \
+            (ZEPHYR_MCUBOOT_APP_0_PRIMARY_SLOT_ID + (x - x)) 
+#define FLASH_AREA_IMAGE_SECONDARY(x)                           \
+            (ZEPHYR_MCUBOOT_APP_0_PRIMARY_SLOT_ID + (x - x))
```
Comment (nordicjm) [inline boot/zephyr/include/sysflash/sysflash.h:59]: Is x needed here at all?
i.e.
```suggestion
            (ZEPHYR_MCUBOOT_APP_0_PRIMARY_SLOT_ID) 
#define FLASH_AREA_IMAGE_SECONDARY(x)                           \
            (ZEPHYR_MCUBOOT_APP_0_PRIMARY_SLOT_ID)
```

**Comment by nordicjm** [comment]
n.b. CI failure is benign, manually tested when used with https://github.com/zephyrproject-rtos/zephyr/pull/53713 and CI all passes

File: boot/zephyr/include/sysflash/sysflash.h (hunk 2 above)
Comment (de-nordic) [reply boot/zephyr/include/sysflash/sysflash.h:59]: This is to prevent variable passed as `x` to be detected as "unused".

**Comment by de-nordic** [comment]
> n.b. CI failure is benign, manually tested when used with [zephyrproject-rtos/zephyr#53713](https://github.com/zephyrproject-rtos/zephyr/pull/53713) and CI all passes

Unfortunately I do not know how to make MCUboot PR build against specific Zephyr PR.

File: boot/zephyr/include/sysflash/sysflash.h (hunk 1 above)
Comment (de-nordic) [reply boot/zephyr/include/sysflash/sysflash.h:54]: Will fix.

**Review by nordicjm — APPROVED** [review]

**Pull request closed without merging**